### ISSUE TITLE
Some files have wrong size attribute

### ISSUE THREAD

**Issue by olevitt** [minio/minio#19198]
Hi ! 

We have some files that have wrong size attribute. The file itself is fine, data is fine and able to be retrieved properly individually but the size attribute is wrong which leads to some client fails (mc client and others) with checks apparently based on this attribute : "Input reader closed pre-maturely. Expected `217739` bytes, but only received `216525` bytes."

Getting size attribute `217739` : 
```
gon@laboitemagique:~$ mc ls sspcloud/projet-ape/log_files/preprocessed/date=2024-01-29/sourceAppel=WF/part-0.parquet --json
{
 "status": "success",
 "type": "file",
 "lastModified": "2024-02-29T04:01:30.737+01:00",
 "size": 217739,
 "key": "part-0.parquet",
 "etag": "ac592efff32cbb13c1a2305b3a7847ba",
 "url": "<redacted>/projet-ape/log_files/preprocessed/date=2024-01-29/sourceAppel=WF/part-0.parquet",
 "versionOrdinal": 1,
 "storageClass": "STANDARD"
}
```

Retrieving the file gets the correct Content-length `216525` :  
```
mc: <DEBUG> GET /projet-ape/log_files/preprocessed/date%3D2024-01-29/sourceAppel%3DWF/part-0.parquet HTTP/1.1
[...]

mc: <DEBUG> HTTP/1.1 200 OK
Content-Length: 216525
```

Single `mc cp` call apparently does not rely on the size attribute so it works fine, I made some checksums checks on the file and it's the exact same as the original - pre upload to minIO - file.
Recursive `mc cp -r` apprently relies on the size attribute and fails with `Input reader closed pre-maturely. Expected `217739` bytes, but only received `216525` bytes.` although it properly downloaded the correct amount of bytes which is ankward.

Not sure how to reproduce the attribute corruption (may be related to #19179 as it's the same cluster).

## Your Environment
* Server is running `minio/minio:RELEASE.2024-03-05T04-48-44Z` on 3 pools.
* Client (`mc`) : `RELEASE.2024-02-24T01-33-20Z` but I tested older versions and alternative S3 clients and have the same issue

EDIT : added full debug traces of `mc cp --debug` and `mc cp -r --debug` for more info : [minio-bug.tar.gz](https://github.com/minio/minio/files/14496870/minio-bug.tar.gz)

**Comment by harshavardhana** [comment]
@olevitt this is a remnant of a rebalance situation. 

First of all you should never have to rebalance on such tiny pools. Rebalance is mostly a useless operation in most environments.

I think we should document this as such as how such old school ideas of rebalance must die, we implemented it grudgingly to avoid other competitors creating FUD.

However there is never a good reason to rebalance.

**Comment by harshavardhana** [comment]
@olevitt can you collect this file from all the disks and share it here along with `format.json` from all the drives at `/mount/.minio.sys/format.json` ?

**Comment by harshavardhana** [comment]
Please collect from all pools

**Comment by olevitt** [comment]
> @olevitt can you collect this file from all the disks and share it here along with `format.json` from all the drives at `/mount/.minio.sys/format.json` ?

To confirm, do you mean the `xl.meta` for this file from each drive from each pool and `format.json` from each drive from each pool ?

**Comment by harshavardhana** [comment]
all of them xl.meta, their constituent parts and format.json @olevitt from all pools, all drives on all nodes.

**Comment by olevitt** [comment]
Here it is : zip file with all format.json and xl.meta (when present) for each server, each drive.
Note that boss4,boss5,boss6,boss7 belongs to pool0, boss8,boss9,boss10,boss11 belongs to pool1 and boss12,boss13,boss14,boss15 belongs to pool2 (the most recent one, added just before the rebalancing)

[export.zip](https://github.com/minio/minio/files/14508900/export.zip)

Let me know if you need anything else

**Comment by harshavardhana** [comment]
Sure thanks // cc @klauspost another variant.

**Comment by harshavardhana** [comment]
fixed by #19227
